docs/swupd_signature_verification.md

```diff
@@ -0,0 +1,113 @@
+Download (if needed) the MoM (Manifest of Manifests) of version V of Clear Linux and its correspondent signature
+Check if MoM signature was signed with swupd root certificate installed on the system.
+Download (if needed) the Manifest of bundle B and calculate its hash. Check if the listed hash for bundle B in MoM is correct.
+Download (if needed) file F and calculate its hash. Check if the listed hash for file F in bundle B Manifest is correct.
+Install the file.
+
+The hash used for that operation is a SHA256 of the content and attributes of the file in question (F or Manifest of bundle B). Swupd hashdump command can be used to calculate the hash for debugging purposes.
+
+## MoM signature verification
+
+The Manifest.MoM file used by swupd should have a corresponding PKCS7 file with the suffix .sig that is going to be used to validate the Manifest.MoM integrity. Swupd uses certificates on the certificate store located on /usr/share/clear/update-ca/Swupd_Root.pem to validate the signature in the PKCS7 bundle. Chained certificates inside the PKCS7 bundle is supported if they are signed by any certificate in the Swupd_Root.pem store.
+
+The openssl command line that is equivalent to the operation performed by swupd is:
+
+openssl smime -verify -in Manifest.MoM.sig -inform DER -content Manifest.MoM -purpose any -CAfile /usr/share/clear/update-ca/Swupd_Root.pem
+
+## Build time parameters
+
+ - Disable the signature verification (--disable-signature-verification):
+Don’t download .sig files and always assume MoM integrity isn’t compromised. Hashes for Bundle Manifests and files are still going to be performed.
+**Warning**: This is not recommended and all swupd operations that install files are going to be insecure.
+
+ - Change certificate path (--with-certpath=PATH)
+Change to PATH the default pem file to be used on MoM signature verification. Default value is /usr/share/clear/update-ca/Swupd_Root.pem.
+
+## Runtime parameters
+
+ - Change certificate path (-C, --certpath)
+Change in runtime to PATH the default pem file to be used on MoM signature verification. The pem file on PATH will be used instead of the default pem file defined in compile time.
+
+ - Disable signature verification (-n, --nosigcheck)
+Proceed even if signature verification fails. Warns the user about that if signature is invalid.
+**Warning**: This is not recommended and all swupd operations that install files are going
+to be insecure.
+
+
+## Examples
+
+Examples on how to create and keep keys to be used with swupd. This command lines are examples only and we suggest you to check openssl documentation on how to create keys with the level of security expected for your use case.
+
+All examples listed below uses a 1 day expiry date which is unrealistic for a real use case. The only purpose of this documentation is to explain what’s supported by swupd and shouldn’t be used as a guide on how to sign Clear Linux based distributions.
```

Comment:
I think the word "uses" here "All examples listed below uses a 1 day expiry" should be use since referring to many examples.